app/assets/javascripts/discourse/controllers/insert-hyperlink.js.es6

```diff
@@ -1,15 +1,98 @@
 import ModalFunctionality from "discourse/mixins/modal-functionality";
+import { searchForTerm } from "discourse/lib/search";
+import { observes } from "ember-addons/ember-computed-decorators";
 
 export default Ember.Controller.extend(ModalFunctionality, {
-  linkUrl: "",
-  linkText: "",
-
   onShow() {
-    Ember.run.next(() =>
-      $(this)
-        .find("input.link-url")
-        .focus()
-    );
+    this.set("linkUrl", "");
+    this.set("linkText", "");
+    this.set("searchResults", []);
+    this.set("selectedRow", -1);
+
+    Ember.run.next(() => {
```

Comment:
Can you explain why you use next and not afterRender here ?

Reply:
No reason, that call to next was already there. But replacing it with `afterRender` makes sense.